pkg/acquisition/modules/syslog/syslog.go

```diff
@@ -0,0 +1,228 @@
+}
+
+func validateAddr(addr string) bool {
+	return net.ParseIP(addr) != nil
+}
+
+func (s *SyslogSource) Configure(yamlConfig []byte, logger *log.Entry) error {
+	s.logger = logger
+	s.logger.Infof("Starting syslog datasource configuration")
+	syslogConfig := SyslogConfiguration{}
+	syslogConfig.Mode = configuration.TAIL_MODE
+	err := yaml.UnmarshalStrict(yamlConfig, &syslogConfig)
+	if err != nil {
+		return errors.Wrap(err, "Cannot parse syslog configuration")
+	}
+	if syslogConfig.Addr == "" {
+		syslogConfig.Addr = "127.0.0.1" //do we want a usable or secure default ?
+	}
+	if syslogConfig.Port == 0 {
+		syslogConfig.Port = 514
+	}
+	if syslogConfig.Proto == "" {
+		syslogConfig.Proto = "udp"
+	}
+	if !validatePort(syslogConfig.Port) {
+		return fmt.Errorf("invalid port %d", syslogConfig.Port)
+	}
+	if !validateAddr(syslogConfig.Addr) {
+		return fmt.Errorf("invalid listen IP %s", syslogConfig.Addr)
+	}
+	s.config = syslogConfig
+	return nil
+}
+
+func (s *SyslogSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	c := make(chan syslogserver.SyslogMessage)
+	s.server = &syslogserver.SyslogServer{Logger: s.logger.WithField("syslog", "internal")}
+	s.server.SetChannel(c)
+	err := s.server.Listen(s.config.Addr, s.config.Port)
+	if err != nil {
+		return errors.Wrap(err, "could not start syslog server")
+	}
+	s.server.StartServer()
+	t.Go(func() error {
+		defer types.CatchPanic("crowdsec/acquis/syslog/live")
+		return s.handleSyslogMsg(out, t, c)
+	})
+	return nil
+}
+
+func (s *SyslogSource) buildLogFromSyslog(ts *time.Time, hostname *string,
+	appname *string, pid *string, msg *string) (string, error) {
+	ret := ""
+	if ts != nil {
+		ret += ts.Format("Jan 2 15:04:05")
+	} else {
+		ret += time.Now().Format("Jan 2 15:04:05")
+	}
+	if hostname != nil {
+		ret += " " + *hostname
+	} else {
+		ret += " unknownhost"
+	}
+	if appname != nil {
+		ret += " " + *appname
+	} else {
+		return "", errors.Errorf("missing appname field in syslog message")
+	}
+	if pid != nil {
+		/*
+			!!! ugly hack !!!
+			Due to a bug in the syslog parser we use (https://github.com/influxdata/go-syslog/issues/31),
+			the ProcID field will contain garbage if the message as a ] anywhere in it.
+			Assume that a correctly formated ProcID only contains number, and if this is not the case, set it to an arbitrary value
+		*/
+		_, err := strconv.Atoi(*pid)
+		if err != nil {
+			ret += "[1]: "
+		} else {
+			ret += "[" + *pid + "]: "
+		}
+	} else {
+		ret += ": "
+	}
+	if msg != nil {
+		ret += *msg
+	} else {
+		return "", errors.Errorf("missing message field in syslog message")
+	}
+	return ret, nil
+
+}
+
+func (s *SyslogSource) handleSyslogMsg(out chan types.Event, t *tomb.Tomb, c chan syslogserver.SyslogMessage) error {
+	for {
+		select {
+		case <-t.Dying():
+			s.logger.Info("Syslog datasource is dying")
+			err := s.server.KillServer()
+			return err
+		case syslogLine := <-c:
+			var line string
+			var ts time.Time
+
```

Comment:
can we have prom metrics tracking events per source (ie. network client ip addr etc.)